src/peer_digest.cc

```diff
@@ -600,24 +572,15 @@ static int
 peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const char *step_name)
 {
     static const SBuf hostUnknown("<unknown>"); // peer host (if any)
-    SBuf host = hostUnknown;
 
-    PeerDigest *pd = nullptr;
-    const char *reason = nullptr;  /* reason for completion */
+    const auto pd = fetch->pd.get();
+    const char *reason = pd ? nullptr : "peer digest disappeared?!";  /* reason for completion */
+    const auto host = pd ? pd->host : hostUnknown;
     const char *no_bug = nullptr;  /* successful completion if set */
-    const int pdcb_valid = cbdataReferenceValid(fetch->pd);
-    const int pcb_valid = pdcb_valid && fetch->pd->peer.valid();
 
     /* test possible exiting conditions (the same for most steps!)
      * cases marked with '?!' should not happen */
 
-    if (!reason) {
-        if (!pdcb_valid || !(pd = fetch->pd))
-            reason = "peer digest disappeared?!";
```

Comment:
Instead of moving this digest validity check logic into `pd`, `reason`, and `host` initialization code above, please preserve this `if` statement. It is best to do all these checks in a similar fashion, to emphasize that each next check relies on the success of the previous one (or a non-nil reason indicating earlier failure). Using this strategy also reduces the diff AFAICT.